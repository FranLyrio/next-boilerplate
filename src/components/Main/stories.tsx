import { ComponentMeta, ComponentStory } from '@storybook/react';
import Main from '.';

export default {
	title: 'Main',
	component: Main
} as ComponentMeta<typeof Main>;

export const Basic: ComponentStory<typeof Main> = (args) => <Main {...args} />;
Basic.args = {
	title: 'Welcome!',
	description:
		'A simple project starter to work with TypeScript, React, Next.js and Styled Components'
};
