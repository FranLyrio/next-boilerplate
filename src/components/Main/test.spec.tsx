import { render, screen } from '@testing-library/react';

import Main from '.';

const props = {
	title: 'Title',
	description: 'Description'
};

describe('<Main />', () => {
	it('should render the heading', () => {
		const { container } = render(<Main {...props} />);

		const heading = screen.getByRole('heading', { name: props.title });

		expect(heading).toBeInTheDocument();
		expect(container.firstChild).toMatchSnapshot();
	});

	it('should render the colors correctly', () => {
		const { container } = render(<Main {...props} />);

		expect(container.firstChild).toHaveStyle({ backgroundColor: '#06092b' });
	});
});
