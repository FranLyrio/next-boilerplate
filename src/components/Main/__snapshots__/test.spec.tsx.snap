// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  font-size: 2.5rem;
}

.c2 {
  font-size: 2rem;
  font-weight: 400;
}

.c3 {
  margin-top: 3rem;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Title
  </h1>
  <h2
    class="c2"
  >
    Description
  </h2>
  <div
    style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
  >
    <div
      style="box-sizing: border-box; display: block; max-width: 100%;"
    >
      <img
        alt=""
        aria-hidden="true"
        role="presentation"
        src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjAwIiBoZWlnaHQ9IjIwMCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
        style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
      />
    </div>
    <img
      alt="Um desenvolvedor de frente para uma tebla com cÃ³digo"
      class="c3"
      decoding="async"
      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
      style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
    />
  </div>
</main>
`;
